src/vmspawn/vmspawn.c

```diff
@@ -474,20 +488,121 @@ static int cmdline_add_vsock(char ***cmdline, int vsock_fd) {
         assert(addr_len >= sizeof addr.vm);
         assert(addr.vm.svm_family == AF_VSOCK);
 
-        log_info("Using vsock-stream:%u:%u", (unsigned) VMADDR_CID_HOST, addr.vm.svm_port);
         r = strv_extendf(cmdline, "type=11,value=io.systemd.credential:vmm.notify_socket=vsock-stream:%u:%u", (unsigned) VMADDR_CID_HOST, addr.vm.svm_port);
         if (r < 0)
                 return r;
 
         return 0;
 }
 
+static int create_runtime_tempdir(const char *our_runtime_dir, const char *template, char **ret_tempdir, char **ret_runtime_directory_property) {
+        _cleanup_(rm_rf_physical_and_freep) char *tempdir = NULL;
+        _cleanup_free_ char *tempdir_template = NULL, *runtime_directory_property = NULL;
+        int r;
+
+        assert(our_runtime_dir);
+        assert(template);
+
+        /* Creates a new temporary directory using the given template
+         *
+         * ret_tempdir is the full path to the tempdir
+         * ret_runtime_directory_property is the tempdir formatted for a unit's RuntimeDirectory= key
+         */
+
+        tempdir_template = path_join(our_runtime_dir, template);
+        if (!tempdir_template)
+                return -ENOMEM;
+
+        r = mkdtemp_malloc(tempdir_template, &tempdir);
```

Comment:
i don't understand why this uses a random path? why not derive this from the service name?

Reply:
Yeah thats a fair point, in my head a path lib $RUNTIME_DIRECTORY/<scope>/<unit name>/ makes sense so I'll do that.